src/services/gapi.js

```diff
@@ -28,9 +31,17 @@ export const getGapi = once(async() => {
   await gapi.client.init({
     apiKey: config.firebaseApiKey,
     clientId: config.firebaseClientId,
-    discoveryDocs: ['https://www.googleapis.com/discovery/v1/apis/classroom/v1/rest'],
-    scope: 'https://www.googleapis.com/auth/classroom.courses.readonly https://www.googleapis.com/auth/classroom.coursework.students https://www.googleapis.com/auth/classroom.coursework.me',
+    discoveryDocs: DISCOVERY_DOCS,
+    scope: GOOGLE_SCOPES_STRING,
   });
 
   return gapi;
 });
+
+export function getGapiSync() {
+  try {
+    return window.gapi;
+  } catch (e) {
+    return null;
```

Comment:
Is this okay or will we need more specific error handling?

Reply:
I think this is kind of backwards—simply referencing `window.gapi` will not itself throw an error, but we _do_ want the function to throw an error if `gapi` is not there. So I would do something like:

```js
if ('gapi' in window) {
  return window.gapi;
}
throw new Error('Attempted to synchronously access `gapi` before it was loaded');
```